micromamba.el

```diff
@@ -235,5 +343,52 @@ full paths."
     (setq micromamba-env-current-prefix nil)
     (run-hooks 'micromamba-postdeactivate-hook)))
 
+;;;###autoload
+(defun micromamba-env-activate-for-buffer ()
+  "Activate the conda environment implied by the current buffer.
+
+This can be set by a buffer-local or project-local variable (e.g. a
+`.dir-locals.el` that defines `conda-project-env-path`), or inferred from an
+`environment.yml` or similar at the project level."
+  (interactive)
+  (let* ((inferred-env (micromamba--infer-env-from-buffer))
+         (env-path (cond
+                    ((bound-and-true-p conda-project-env-path) conda-project-env-path)
+                    ((not (eql inferred-env nil)) (micromamba-env-name-to-dir inferred-env))
+                    (t nil))))
+
+    (if (not (eql env-path nil))
+        (micromamba-activate env-path)
+      (if micromamba-message-on-environment-switch
+          (message "No Conda environment found for <%s>" (buffer-file-name))))))
+
+(defun micromamba--switch-buffer-auto-activate (&rest args)
+  "Add Conda environment activation if a buffer has a file, handling ARGS."
+  (let ((filename (buffer-file-name)))
+    (when filename
+      ;; (message "switch-buffer auto-activating on <%s>" filename)
+      (with-demoted-errors "Error: %S"
+        (micromamba-env-activate-for-buffer)))))
+
+;;;###autoload
+(define-minor-mode micromamba-env-autoactivate-mode
+  "Toggle conda-env-autoactivate mode.
+
+This mode automatically tries to activate a conda environment for the current
+buffer."
+  ;; The initial value.
+  :init-value nil
+  ;; The indicator for the mode line.
+  :lighter nil
+  ;; The minor mode bindings.
+  :keymap nil
+  ;; Kwargs
+  :group 'micromamba
+  :global t
+  ;; Forms
+  (if micromamba-env-autoactivate-mode ;; already on, now switching off
+      (advice-add 'switch-to-buffer :after #'micromamba--switch-buffer-auto-activate)
```

Comment:
Emacs 27.1 introduced a hook called `window-selection-change-functions`, I suggest using that instead of `advice-add`.